packages/headless/src/controllers/search-box/headless-search-box.test.ts

```diff
@@ -165,14 +166,25 @@ describe('headless searchBox', () => {
     });
   });
 
-  it(`when calling selectSuggestion
-    should dispatch a selectQuerySuggestion action`, () => {
-    const value = 'i like this expression';
-    searchBox.selectSuggestion(value);
+  describe('#selectSuggestion', () => {
+    it('dispatches a selectQuerySuggestion action', () => {
+      const value = 'i like this expression';
+      searchBox.selectSuggestion(value);
 
-    expect(engine.actions).toContainEqual(
-      selectQuerySuggestion({id, expression: value})
-    );
+      expect(engine.actions).toContainEqual(
+        selectQuerySuggestion({id, expression: value})
+      );
+    });
+
+    it('dispatches executeSearch with a logQuerySuggestionClick search event', () => {
+      const suggestion = 'a';
+      searchBox.selectSuggestion(suggestion);
+
+      const action = engine.findAsyncAction(executeSearch.pending);
+      expect(action!.meta.arg.toString()).toBe(
+        logQuerySuggestionClick({id, suggestion}).toString()
+      );
```

Comment:
Serializing the `arg` function on the thunk, and comparing it do the serialized analytics function was the simplest way I found to check that the correct analytics action was being passed to `executeSearch`.